File: perl/CHAP1/04.pl
#!/usr/bin/perl

use strict;
use warnings;

# . を削除し, スペース毎に分割して各単語毎にリスト化
my $str = "Hi He Lied Because Boron Could Not Oxidize Fluorine. New Nations Might Also Sign Peace Security Clause. Arthur King Can.";
$str =~ s/\.//g;
my @word = split(" ", $str);

my %hash = ();
my @num = qw/1 5 6 7 8 9 15 16 19/;
for(my $i = 0; $i < $#word + 1; $i++) {
	my @char = split //, $word[$i];
	if(grep {$_ == $i + 1} @num) {
		$hash{$char[0]} = $i + 1; # 先頭の1文字を key に
	} else {
		$hash{$char[0] . $char[1]} = $i + 1; # 先頭の2文字を key に
	}
}

# 元素番号順 (単語の位置の昇順) に出力
foreach my $key (sort {$hash{$a} <=> $hash{$b}} keys %hash) {
	print "$hash{$key} : $key\n";
}

File: python/CHAP1/04.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-

str = "Hi He Lied Because Boron Could Not Oxidize Fluorine. New Nations Might Also Sign Peace Security Clause. Arthur King Can."
list = str.split()

dict = {}
single = [0, 4, 5, 6, 7, 8, 14, 15, 18]

for i in range(len(list)):
    c_len = 1 if i in single else 2
    dict[list[i][:c_len]] = i + 1

for k, v in sorted(dict.items(), key=lambda x: x[1]):
    print(k, v)
